pkg/api/v1/clients/kube/resource_client_test.go

```diff
@@ -512,11 +520,44 @@ var _ = Describe("Test Kube ResourceClient", func() {
 			BeforeEach(func() {
 				clientset = fake.NewSimpleClientset(v1.MockResourceCrd)
 
-				// Create initial resources
-				Expect(util.CreateMockResource(ctx, clientset, namespace1, "res-1", "val-1")).NotTo(HaveOccurred())
-				Expect(util.CreateMockResource(ctx, clientset, namespace1, "res-2", "val-2")).NotTo(HaveOccurred())
-				Expect(util.CreateMockResource(ctx, clientset, namespace1, "res-3", "val-3")).NotTo(HaveOccurred())
-				Expect(util.CreateMockResource(ctx, clientset, namespace2, "res-4", "val-4")).NotTo(HaveOccurred())
+				metadataForMockResources := []*core.Metadata{
+					{
+						Name:      "res-1",
+						Namespace: namespace1,
+						Labels: map[string]string{
+							"name":      "res-1",
+							"namespace": namespace1,
+						},
+					},
+					{
+						Name:      "res-2",
+						Namespace: namespace1,
+						Labels: map[string]string{
+							"name":      "res-2",
+							"namespace": namespace1,
+						},
+					},
+					{
+						Name:      "res-3",
+						Namespace: namespace1,
+						Labels: map[string]string{
+							"name":      "res-3",
+							"namespace": namespace1,
+						},
+					},
+					{
+						Name:      "res-4",
+						Namespace: namespace2,
+						Labels: map[string]string{
+							"name":      "res-4",
+							"namespace": namespace2,
+						},
+					},
+				}
+
+				for i, meta := range metadataForMockResources {
+					Expect(util.CreateMockResourceWithMetadata(ctx, clientset, meta, fmt.Sprintf("val-%d", i)))
```

Comment:
```suggestion
					Expect(util.CreateMockResourceWithMetadata(ctx, clientset, meta, fmt.Sprintf("val-%d", i))).NotTo(HaveOccurred())
```
should we be checking the error?

Reply:
you're so right!